Built-in libraries from the Postman Sandbox aren't included when running postman sync.

When running `postman sync` from a bootstrapped collection, Browserify can't build the dependency graph if the Postman built-in modules haven't been `npm install`'d.  Can these either be detected ahead of time and installed or can we optimistically install all built-in libraries from the Postman sandbox on the initial bootstrap?

```
➜ postman sync
(node:32473) UnhandledPromiseRejectionWarning: Error: Can't walk dependency graph: Cannot find module 'lodash' from '/private/tmp/postman-cli-test/postman/Utilities/Delay Response/test.js'
    required by /private/tmp/postman-cli-test/postman/Utilities/Delay Response/test.js
    at /usr/local/lib/node_modules/@matt.ball/postman-cli/node_modules/resolve/lib/async.js:115:35
    at processDirs (/usr/local/lib/node_modules/@matt.ball/postman-cli/node_modules/resolve/lib/async.js:268:39)
    at isdir (/usr/local/lib/node_modules/@matt.ball/postman-cli/node_modules/resolve/lib/async.js:275:32)
    at /usr/local/lib/node_modules/@matt.ball/postman-cli/node_modules/resolve/lib/async.js:25:69
    at FSReqCallback.oncomplete (fs.js:175:21)
```

Note: The workaround for this was just doing `npm i lodash` and moving forward.

--- a/package.json
+++ b/package.json
@@ -25,6 +25,7 @@
     "browserify": "^16.5.0",
     "chalk": "^2.4.2",
     "commander": "^4.0.1",
+    "detective": "^5.2.0",
     "enquirer": "^2.3.2"
   },
   "devDependencies": {

--- a/src/sync.js
+++ b/src/sync.js
@@ -49,11 +49,8 @@ async function maybeBundle (path) {
 
 async function bundle (path) {
   const b = browserify()
+  b.external(sandboxBuiltins)
   b.add(path)
-
-  for (const builtin of sandboxBuiltins) {
-    b.external(builtin)
-  }
 
   const doBundle = promisify(b.bundle.bind(b))
   const buf = await doBundle()

--- a/src/sync.js
+++ b/src/sync.js
@@ -1,9 +1,19 @@
 const fs = require('fs')
 const browserify = require('browserify')
+const detective = require('detective')
 const promisify = require('util').promisify
 const config = require('./lib/config')
 const file = require('./lib/file')
 const recurseCollection = require('./lib/recurse-collection')
+
+// Reference: https://learning.postman.com/docs/writing-scripts/script-references/postman-sandbox-api-reference/
+const sandboxBuiltins = [
+  'ajv', 'atob', 'btoa', 'chai', 'cheerio', 'crypto-js',
+  'csv-parse/lib/sync', 'lodash', 'moment', 'postman-collection',
+  'tv4', 'uuid', 'xml2js', 'path', 'assert', 'buffer', 'util',
+  'url', 'punycode', 'querystring', 'string-decoder', 'stream',
+  'timers', 'events'
+]
 
 module.exports = async function sync () {
   const collection = await recurseCollection(mapFileToItem)
@@ -21,7 +31,7 @@ async function mapFileToItem (req, context = '', type) {
   if (localFileExists) {
     if (isScript) {
       const index = req.event.findIndex((el) => el.listen === type)
-      req.event[index].script.exec = await bundle(path)
+      req.event[index].script.exec = await maybeBundle(path)
     } else {
       req[type] = readItemFile(path)
     }
@@ -30,18 +40,16 @@ async function mapFileToItem (req, context = '', type) {
   return req
 }
 
+async function maybeBundle (path) {
+  const source = fs.readFileSync(path)
+  const requires = detective(source).filter(r => !sandboxBuiltins.includes(r))
+
+  return requires.length ? bundle(path) : source.toString('utf8')
+}
+
 async function bundle (path) {
   const b = browserify()
   b.add(path)
-
-  // Reference: https://learning.postman.com/docs/writing-scripts/script-references/postman-sandbox-api-reference/
-  const sandboxBuiltins = [
-    'ajv', 'atob', 'btoa', 'chai', 'cheerio', 'crypto-js',
-    'csv-parse/lib/sync', 'lodash', 'moment', 'postman-collection',
-    'tv4', 'uuid', 'xml2js', 'path', 'assert', 'buffer', 'util',
-    'url', 'punycode', 'querystring', 'string-decoder', 'stream',
-    'timers', 'events'
-  ]
 
   for (const builtin of sandboxBuiltins) {
     b.external(builtin)

--- a/src/sync.js
+++ b/src/sync.js
@@ -34,6 +34,19 @@ async function bundle (path) {
   const b = browserify()
   b.add(path)
 
+  // Reference: https://learning.postman.com/docs/writing-scripts/script-references/postman-sandbox-api-reference/
+  const sandboxBuiltins = [
+    'ajv', 'atob', 'btoa', 'chai', 'cheerio', 'crypto-js',
+    'csv-parse/lib/sync', 'lodash', 'moment', 'postman-collection',
+    'tv4', 'uuid', 'xml2js', 'path', 'assert', 'buffer', 'util',
+    'url', 'punycode', 'querystring', 'string-decoder', 'stream',
+    'timers', 'events'
+  ]
+
+  for (const builtin of sandboxBuiltins) {
+    b.external(builtin)
+  }
+
   const doBundle = promisify(b.bundle.bind(b))
   const buf = await doBundle()
   const script = buf.toString()

